### PR TITLE
Detect error slot for inputs instead of hasErrors prop

### PR DESCRIPTION
**Pull request by benjaminknox**
Resolves https://github.com/brave/leo/issues/455

Currently to show error messages on inputs, `hasErrors` prop is passed. This uses `$$slots.errors` instead of exposing the `hasErrors` prop, implemented on `Dropdown` and `Input` components.

### With slot present
Dropdown:
Screenshot 2023-10-30 at 11 05 58 AM

Input:
Screenshot 2023-10-30 at 11 05 36 AM

### Without slot present

Dropdown:
image

Input:
image

### CODE REVIEW

File: src/components/input/input.svelte [hunk 1]
```diff
@@ -164,7 +156,7 @@
   </slot>
   <slot slot="label" />
 </FormItem>
-{#if showErrors && hasErrors}
+{#if showErrors && $$slots.errors}
```
Comment (fallaciousreasoning) [inline src/components/input/input.svelte:159]: I don't think we should need the `$$slots.errors` here - it's like saying, if we don't have the slot, don't show the slot (which won't happen anyway)
```suggestion
{#if showErrors}
```

File: src/components/dropdown/dropdown.svelte [hunk 2]
```diff
@@ -103,7 +102,7 @@
     <slot />
   </Menu>
 </div>
-{#if showErrors && hasErrors}
+{#if showErrors && $$slots.errors}
```
Comment (fallaciousreasoning) [inline src/components/dropdown/dropdown.svelte:105]:
```suggestion
{#if showErrors}
```

I don't think we need the `$$slots.errors` check :smile:

**Review by fallaciousreasoning — COMMENTED** [review]
This looks great! Couple of minor changes and we should be good!

File: src/components/input/input.svelte (hunk 1 above)
Comment (benjaminknox) [reply src/components/input/input.svelte]: Ah, yes thanks for pointing that out, makes sense

**Review by fallaciousreasoning — APPROVED** [review]
LGTM!

**Pull request merged**